lib/vkloader.c

```diff
@@ -749,20 +749,29 @@ linearTilingPadCallback(int miplevel, int face,
  * limited number of formats and features. Generally @c VK_IMAGE_TILING_OPTIMAL
  * is preferred. The latter requires a staging buffer so will use more memory
  * during loading.
+ * 
+ * If a pointer to a thread-safe set of suballocator callbacks is provided, they
+ * will be used instead of manual allocation of a VkDeviceMemory. A 64 bit uint
+ * is returned on memory procurement that references the suballocated page(s).
  *
- * @param[in] This          pointer to the ktxTexture from which to upload.
- * @param [in] vdi          pointer to a ktxVulkanDeviceInfo structure providing
- *                          information about the Vulkan device onto which to
- *                          load the texture.
- * @param [in,out] vkTexture pointer to a ktxVulkanTexture structure into which
- *                           the function writes information about the created
- *                           VkImage.
- * @param [in] tiling       type of tiling to use in the destination image
- *                          on the Vulkan device.
- * @param [in] usageFlags   a set of VkImageUsageFlags bits indicating the
- *                          intended usage of the destination image.
- * @param [in] finalLayout  a VkImageLayout value indicating the desired
- *                          final layout of the created image.
+ * @param[in] This                        pointer to the ktxTexture from which to upload.
+ * @param [in] vdi                        pointer to a ktxVulkanDeviceInfo structure providing
+ *                                        information about the Vulkan device onto which to
+ *                                        load the texture.
+ * @param [in,out] vkTexture              pointer to a ktxVulkanTexture structure into which
+ *                                        the function writes information about the created
+ *                                        VkImage.
+ * @param [in] tiling                     type of tiling to use in the destination image
+ *                                        on the Vulkan device.
+ * @param [in] usageFlags                 a set of VkImageUsageFlags bits indicating the
+ *                                        intended usage of the destination image.
+ * @param [in] finalLayout                a VkImageLayout value indicating the desired
+ *                                        final layout of the created image.
+ * @param [in] subAllocatorCallbacks      pointer to a potential set of thread-safe 
```

Comment:
pointer to a set of thread-safe suballocator callbacks that wrap around suballocator calls: alloc, bindbuffer, bindimage, map, unmap and free. They use a uint64_t stored in the @c allocationId field of the structure pointed at by @a vkTexture to reference allocated page(s).

Reply:
Ok, got it! I guess the @A and @c have meaning here.

Reply:
@a means display the word in italics as it is one of the function arguments. That @a became @A is the result of some autocorrect I didn't notice before. @c means display the word in typewriter font as it is code. See https://www.doxygen.nl/manual/commands.html#cmda.